genie-agent/src/main/java/com/netflix/genie/agent/cli/JobRequestConverter.java

```diff
@@ -75,5 +85,33 @@ public AgentJobRequest agentJobRequestArgsToDTO(
             .withTimeout(jobRequestArguments.getTimeout())
             .withRequestedId(jobRequestArguments.getJobId())
             .build();
+
+        final Set<ConstraintViolation<AgentJobRequest>> violations = validator.validate(agentJobRequest);
+
+        if (!violations.isEmpty()) {
+            throw new ConversionException(violations);
+        }
+
+        return agentJobRequest;
+    }
+
+    /**
+     * Exception thrown in case of conversion error due to resulting object failing validation.
+     */
+    public static final class ConversionException extends Exception {
```

Comment:
Seems like an exception you might want to use in more places but I guess this is fine for now